rcljava/src/main/cpp/org_ros2_rcljava_RCLJava.cpp

```diff
@@ -58,20 +59,73 @@ Java_org_ros2_rcljava_RCLJava_nativeCreateNodeHandle(
 
   rcl_context_t * context = reinterpret_cast<rcl_context_t *>(context_handle);
 
+  // TODO(clalancette): we could probably make this faster by just doing these
+  // method lookups during some initialization time.  But we'd have to add a lot
+  // more infrastructure for that, and we don't expect to call
+  // 'nativeCreateNodeHandle' that often, so I think this is OK for now.
+  jclass java_util_ArrayList =
+    static_cast<jclass>(env->NewGlobalRef(env->FindClass("java/util/ArrayList")));
+  jmethodID java_util_ArrayList_size = env->GetMethodID(java_util_ArrayList, "size", "()I");
+  jmethodID java_util_ArrayList_get = env->GetMethodID(
+    java_util_ArrayList, "get", "(I)Ljava/lang/Object;");
+
+  rcl_arguments_t arguments = rcl_get_zero_initialized_arguments();
+  char ** argv = NULL;
+  jint argc = env->CallIntMethod(cli_args, java_util_ArrayList_size);
+  argv = static_cast<char **>(malloc(argc * sizeof(char *)));
+  for (jint i = 0; i < argc; ++i) {
+    jstring element =
+      static_cast<jstring>(env->CallObjectMethod(cli_args, java_util_ArrayList_get, i));
+    argv[i] = const_cast<char *>(env->GetStringUTFChars(element, nullptr));
```

Comment:
Use [DeleteLocalRef](https://docs.oracle.com/javase/8/docs/technotes/guides/jni/spec/functions.html#DeleteLocalRef) to delete `element` after using it.
Java will delete all local references on return, but it's a good practice to manually delete them on loops to avoid the local frame to unnecessarily grow.

Reply:
I didn't do the `DeleteLocalRef` thing because of the complexities of cleanup.  Since I didn't expect there to be tons of arguments anyway, I was just going to let the VM deal with cleaning them up.  But let me see how it goes with a revamp of the cleanup as you mention below.

Reply:
Fixed in dd844a9